src/main/java/ro/msg/learning/shop/configuration/LocationConfiguration.java

```diff
@@ -0,0 +1,24 @@
+package ro.msg.learning.shop.configuration;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.strategy.LocationStrategy;
+import ro.msg.learning.shop.strategy.MostAbundantLocationImpl;
+import ro.msg.learning.shop.strategy.SingleLocationImp;
+import ro.msg.learning.shop.strategy.StrategyTypeEnum;
+
+@Configuration
+public class LocationConfiguration {
+	@Value("${strategy}")
+	StrategyTypeEnum strategy;
+
+	@Bean
+	public LocationStrategy locationStrategy() {
+
+		return switch (strategy) {
+			case MOST_ABUNDANT -> new MostAbundantLocationImpl();
```

Comment:
Usually, best practice for this is to add another case statement for SingleLocation and throw an exception on default case. If the code goes on the default case, then it means that the strategy specified is not implemented in the application. In this way it will not be confusing for the user if he want for example "the closest location" and receives instead the location returned by the SingleLocation strategy.

Reply:
Updated.